### PR TITLE
Create URL field

### PR DESCRIPTION
**Pull request by tintinthong**
Creates URL field 

**This displays the atom in a card**

Screenshot 2024-12-02 at 20 53 38

**Edit mode**

Screenshot 2024-12-02 at 20 57 37

**Invalid URL**

Screenshot 2024-12-02 at 20 57 28

**Valid URL**

Screenshot 2024-12-02 at 21 00 32

**Question**

Shud I put this in base?

### CODE REVIEW

File: packages/experiments-realm/url.gts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  };
+}
+
+function isValidUrl(urlString: string): boolean {
+  try {
+    new URL(urlString);
+    return true;
+  } catch (err) {
+    return false;
+  }
```
Comment (backspace) [inline packages/experiments-realm/url.gts:60]: `URL.canParse` [does exist](https://caniuse.com/mdn-api_url_canparse_static) for this use.

**Review by backspace — APPROVED** [review]
This seems like something that could be in `boxel-ui`, was there any discussion of that possibility?

**Review by lukemelia — APPROVED** [review]
@tintinthong use canParse per Buck's suggestion. Let's discuss whether this should be promoted to boxel-ui on Thursday

**Review by burieberry — COMMENTED** [review]
Looks good with the improvement Buck mentioned. A note on link styling, Gardella mentioned that hover state should be same color but with underline text-decoration. So you can override the default link styling in this component for hover state.

File: packages/experiments-realm/url.gts (hunk 1 above)
Comment (tintinthong) [reply packages/experiments-realm/url.gts:60]: In order for this to work with types, I'd need to upgrade to typescript 5.2. Shall I upgrade our types to the most **latest** version of typescript or upgrade it to 5.2.2? I will merge first and resolve in another PR.

https://github.com/microsoft/TypeScript/issues/55623#issuecomment-1704705924

**Pull request merged**